File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02838/p02838_num3_parsed.cpp
#include <iostream>
#include <vector>

using namespace std;

const int MOD = 1000000007;

int main() {
    int N;
    cin >> N;

    vector<long long> A(N);
    for (int i = 0; i < N; i++) {
        cin >> A[i];
    }

    long long result = 0;
    for (int i = 0; i < 60; i++) {
        long long count = 0;
        for (int j = 0; j < N; j++) {
            if ((A[j] >> i) & 1) {
                count++;
            }
        }
        result += (count * (N - count) % MOD) * (1LL << i) % MOD;
        result %= MOD;
    }

    cout << result << endl;

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02838/p02838_num3_parsed.py
import sys
input = sys.stdin.readline

MOD = 10**9 + 7

def main():
    # Read input
    N = int(input())
    A = list(map(int, input().split()))

    # Calculate XOR sum
    xor_sum = 0
    for i in range(N-1):
        for j in range(i+1, N):
            xor_sum += A[i] ^ A[j]
            xor_sum %= MOD

    # Print result
    print(xor_sum)

if __name__ == '__main__':
    main()